### PR TITLE
340 multivpc grouping

### PR DESCRIPTION
**Pull request by ShiriMoran**
solved #340 
@haim-kermany as far as I can test it seems OK, can you please verify with drawio?

### CODE REVIEW

File: pkg/vpcmodel/output.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ func NewOutputGenerator(c1, c2 map[string]*VPCConfig, grouping bool, uc OutputUs
 			}
 		}
 	}
+	unifyMultiVPC(c1, res.NodesConn, res.SubnetsConn)
```
Comment (haim-kermany) [inline pkg/vpcmodel/output.go:109]: do we need unifyMultiVPC() in archMode?

**Review by haim-kermany — CHANGES_REQUESTED** [review]
Hi
most of the changed code of this PR is functions that you move from one file to the another.
its make it imposable to do an effective code review.
please move the funcs to their  original place for code reviewing
and rearrange the code in a separated PR

File: pkg/vpcmodel/output.go (hunk 1 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/output.go:109]: Probably not - as there is nothing to group in `archMode`.

File: cmd/analyzer/main.go [hunk 2]
```diff
@@ -78,6 +78,7 @@ func analysisVPCConfigs(c1, c2 map[string]*vpcmodel.VPCConfig, inArgs *InArgs, o
 	}
 
 	outFormat := getOutputFormat(inArgs)
+
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply cmd/analyzer/main.go:81]: done

File: pkg/vpcmodel/output.go [hunk 3]
```diff
@@ -48,8 +48,8 @@ type OutputGenerator struct {
 	config2        map[string]*VPCConfig // specified only when analysis is diff
 	outputGrouping bool
 	useCase        OutputUseCase
-	nodesConn      map[string]*VPCConnectivity
-	subnetsConn    map[string]*VPCsubnetConnectivity
+	NodesConn      map[string]*VPCConnectivity
+	SubnetsConn    map[string]*VPCsubnetConnectivity
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/vpcmodel/output.go:52]: right. done

**Comment by ShiriMoran** [comment]
> Hi most of the changed code of this PR is functions that you move from one file to the another. its make it imposable to do an effective code review. please move the funcs to their original place for code reviewing and rearrange the code in a separated PR

here:
https://github.com/np-guard/vpc-network-config-analyzer/pull/423

**Comment by ShiriMoran** [comment]
will reopen once https://github.com/np-guard/vpc-network-config-analyzer/pull/423 is merged

**Pull request closed without merging**